src/stores/activitiesStores/Finalize.js

```diff
@@ -0,0 +1,57 @@
+import { observable, action, runInAction } from 'mobx';
+import _ from 'lodash';
+import { Token, OracleStatus, AppLocation, SortBy } from '../../constants';
+import { queryAllOracles } from '../../network/graphQuery';
+import Oracle from '../models/Oracle';
+
+
+export default class {
+  @observable loading = true
+  @observable loadingMore = false
+  @observable list = []
+  @observable hasMore = true
+  @observable skip = 0
+  limit = 50
+
+  constructor(app) {
+    this.app = app;
+  }
+  // init to fetch the list
+  @action
+  init = async (limit = this.limit) => {
+    if (limit === this.limit) {
+      this.skip = 0;
+    }
+    this.hasMore = true;
+    this.app.ui.location = AppLocation.resultSetting;
+    this.list = await this.fetch(limit);
+    runInAction(() => {
+      this.skip += limit;
+      this.loading = false;
+    });
+  }
+  // load more
+  @action
+  loadMore = async () => {
+    if (this.hasMore) {
+      this.loadingMore = true;
+      this.skip += this.limit;
+      const nextFewEvents = await this.fetch();
+      runInAction(() => {
+        this.list = [...this.list, ...nextFewEvents];
+        this.loadingMore = false;
+      });
+    }
+  }
+  fetch = async (limit = this.limit, skip = this.skip) => {
+    let data = [];
+    const filters = [{ token: Token.Bot, status: OracleStatus.WaitResult }];
+    const orderBy = { field: 'endTime', direction: SortBy.Ascending };
+    if (this.hasMore) {
+      data = await queryAllOracles(filters, orderBy, limit, skip);
+      data = _.uniqBy(data, 'txid').map((oracle) => new Oracle(oracle, this.app));
+      if (data.length < limit) this.hasMore = false;
```

Comment:
yes, after digging into it, I notice we should not check the length of the `data` here(under fetch fucntion) since the data will always be only the same length as limit, because the data is back from `queryAllOracles` which the limit as configuration.

I fix it by adding a `reaction` to constructor.

the method I debugged is to change limit from 50 to 4, too see it loading row by row.

Reply:
at the same time, I also change the default of limit from 50 down to 16, because I think 50 is too many, 16 on a desktop will be 4 rows.

Reply:
leave a bug in the `allEventsStore`(alex's code), already reported to @alex-cory , he will fix it in an another PR.

Reply:
@amazingandyyy There was an issue with the rows not loading more previously that was ensured by upping the number of displayed. If your reaction fixes it then we are good.

But, for the light client version, I feel like we should minimize the number of network calls to the server if we can. ie. make 1 call for 50  vs 3 calls of 16.

Also, another thing I wanted to account for is really large screens like the giant Apple desktop screens. How many events can fit in the row there?